ios/NativeModules/Multipeer.m

```diff
@@ -0,0 +1,88 @@
+#import "Multipeer.h"
+
+@interface Multipeer()
+
+@end
+
+@implementation Multipeer
+
+RCT_EXPORT_MODULE();
+
+RCT_EXPORT_METHOD(init:(NSString *)name address:(NSString *)address avatar:(NSString *)avatar) {
+  [self initialize:name address:address avatar:avatar];
+}
+
+RCT_EXPORT_METHOD(advertiseOn) {
+  [self advertiseToggle:true];
+}
+
+RCT_EXPORT_METHOD(advertiseOff) {
+  [self advertiseToggle:false];
+}
+
+RCT_EXPORT_METHOD(scan) {
+  [self scanStart];
+}
+
+#pragma mark -
+#pragma mark Public Methods
+- (void)initialize:(NSString *)name address:(NSString *)address avatar:(NSString *)avatar {
+  self.name = name;
+  self.address = address;
+  self.avatar = avatar;
+  self.peerID = [[MCPeerID alloc] initWithDisplayName:address];
+  self.session = [[MCSession alloc] initWithPeer:self.peerID];
+  self.session.delegate = self;
+}
+
+- (void)scanStart {
+  self.browser = [[PeerBrowser alloc] init];
+  [self.browser initialize:@"textileio" peerID:self.peerID];
```

Comment:
advertising and browsing is namespaced by textile. this is cool because we could generate namespaces for say, threads, where only others already in the thread might listen for those.